src/components/BaseComponents/TextInput/TextInput.tsx

```diff
@@ -1,28 +1,42 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import FormGroup, {makeErrorId, makeHintId} from '../FormGroup/FormGroup';
+import FormGroup, { makeErrorId, makeHintId } from '../FormGroup/FormGroup';
 
-export default function TextInput(props){
+export default function TextInput(props) {
+  const { name, errorText, hintText, inputProps = {}, maxLength } = props;
 
-  const {name, errorText, hintText, inputProps={}} = props;
+  const inputClasses = `govuk-input ${errorText ? 'govuk-input--error' : ''}`.trim();
 
-  const inputClasses = `govuk-input ${errorText?'govuk-input--error':""}`.trim();
+  const textInputClassesWithWidth = (width: number) => {
+    if (width !== undefined || width !== null) {
+      if (width >= 6 && width <= 10) return `${inputClasses} govuk-input--width-${10}`.trim();
+      else if (width >= 11 && width <= 20) return `${inputClasses} govuk-input--width-${20}`.trim();
+      else return `${inputClasses} govuk-input--width-${width}`.trim();
```

Comment:
Can we have a case for anything above 20? Thinking it should just return the same as line 15

Reply:
For now anything above 20 we are taking the default one